### DISCUSSION TITLE
404 when testing views endpoint (Unit Test)

### DISCUSSION THREAD

**Discussion by klevamane** [django-tenants/django-tenants#1084] [Q&A]
I keep getting 404 when testing view endpoints even though the reverse("path") returns the correct path

```
client.get(correct_view_url_path)
```
always returns 404, i'm really not sure why

**Comment by lip3l** [comment]
reverse() only proves the path is correct. With django-tenants, the tenant is resolved from the request Host. In tests the client usually uses testserver, so the middleware stays on the public schema, and your tenant-only endpoint returns 404.
Fix: send the request with the tenant domain as HTTP_HOST, or use django_tenants.test.client.TenantClient (it sets HTTP_HOST from tenant.get_primary_domain().domain).
After setting the host to the tenant domain, the same reverse() URL should return the expected status. If this solves it, please mark as Accepted Answer
